### PR TITLE
[H2I-6] Make influxdb connection more robust

### PR DESCRIPTION
**Pull request by emiliyank**
Use Asynchronous non-blocking API - WriteApi  https://github.com/influxdata/influxdb-client-java/tree/master/client#asynchronous-non-blocking-api

### CODE REVIEW

File: src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java [hunk 1]
```diff
@@ -2,15 +2,38 @@
...
+        try{
+            return InfluxDBClientFactory.create(influxDBProperties.getUrl(), influxDBProperties.getToken().toCharArray(), influxDBProperties.getOrg(), influxDBProperties.getBucket());
+        } catch (InfluxException e) {
+            log.error("Failed to connect to InfluxDB", e);
+            return null;
+        }
+    }
+
+    //set the connection only if it is empty or not active
+    public InfluxDBClient updateConnectionIfNeeded() {
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java]: `updateConnectionIfNeeded()` should be renamed to `createInfluxClient` or so.

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 2]
```diff
@@ -1,50 +1,46 @@
...
-        return point;
+    public void writePoint(Point point) {
+        measurementPoints.add(point);
     }
 
-    private void updateConnectionIfNeeded() {
-        //set the connection only if it is empty of not active
-        if(influxDBClient == null || !influxDBClient.ping()) {
-            influxDBClient = influxDBConfig.dbConnection();
+    @Scheduled(fixedRate = 30000)
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: `30000` should be constant wir proper name

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 3]
```diff
@@ -1,50 +1,46 @@
...
+    public void persistPointsToInflux() {
+        log.info("persistPointsToInflux() - every 30 secs");
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: imho this will cause too much logs and this logs don't have a real value

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 4]
```diff
@@ -1,50 +1,46 @@
...
+        influxDBClient = influxDBConfig.updateConnectionIfNeeded();
+        int numberOfPointsWritten = 0;
+
+        if(influxDBClient.ping()) {
+            log.info("InfluxDbRepository.writePoint() with successful ping()");
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: do we really need this log output?

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 5]
```diff
@@ -1,50 +1,46 @@
...
+            while (!measurementPoints.isEmpty()) {
+                Point point = measurementPoints.poll();
+                WriteApi writeApi = influxDBClient.makeWriteApi();
+                writeApi.writePoint(point);
+                numberOfPointsWritten++;
+                log.info("writeApi.writePoint() with time: " + point.getTime());
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: do we really need this log output?

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 6]
```diff
@@ -1,50 +1,46 @@
...
-                    .addField("totalConsumption(1_8_0)", sensorEntry.getSml().getTotalConsumption())
-                    .addField("tariff1Consumption(1_8_1)", sensorEntry.getSml().getTariff1Consumption())
-                    .addField("tariff2Consumption(1_8_2)", sensorEntry.getSml().getTariff2Consumption())
-                    .addField("energyExport(2_8_0)", sensorEntry.getSml().getEnergyExport())
-                    .addField("currentConsumption(16_7_0)", sensorEntry.getSml().getCurrentConsumption())
-                    .addField("currentConsumptionPhase1(36_7_0)", sensorEntry.getSml().getCurrentConsumptionPhase1())
-                    .addField("currentConsumptionPhase2(56_7_0)", sensorEntry.getSml().getCurrentConsumptionPhase2())
-                    .addField("currentConsumptionPhase3(76_7_0)", sensorEntry.getSml().getCurrentConsumptionPhase3())
-                    .time(Instant.now(), WritePrecision.MS);
+        log.info("writePoint() wrote : " + numberOfPointsWritten + " point(s).");
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: do we really need this log output?

File: src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java [hunk 7]
```diff
@@ -14,21 +16,35 @@
...
-                if(receivedMessage.contains("1_8_0")) {
-                    SensorEntry sensorEntry = objectMapper.readValue(receivedMessage, SensorEntry.class);
-                    influxDbRepository.writePoint(sensorEntry);
-                }
-            } catch (JsonProcessingException e) {
-                log.error("objectMapper JsonProcessingException" + e);
+    @Override public void messageArrived(String s, MqttMessage mqttMessage) {
+        try {
+            String receivedMessage = new String(mqttMessage.getPayload());
+            log.info("Received message: " + receivedMessage);
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java]: do we really need this log output?

File: src/test/java/de/wejago/hichi2influx/unit/InfluxDbRepositoryTest.java [hunk 8]
```diff
@@ -24,39 +26,33 @@
...
-        Point actualPoint = influxDbRepository.writePoint(sensorEntry);
-
-        //THEN
-        Point expectedPoint = generateExpectedMeasurementPoint(sensorEntry);
-        verify(influxDBClient, times(1)).getWriteApiBlocking();
-        assertThat(actualPoint).usingRecursiveComparison()
-            .ignoringFields("time")
-            .isEqualTo(expectedPoint);
-    }
+//    @Test
```
Comment (cupakob) [inline src/test/java/de/wejago/hichi2influx/unit/InfluxDbRepositoryTest.java]: remove unused code

File: src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java [hunk 9]
```diff
@@ -2,15 +2,38 @@
...
 @Configuration
 @RequiredArgsConstructor
+@Slf4j
 public class InfluxDBConfig {
     private final InfluxDBProperties influxDBProperties;
+    private InfluxDBClient influxDBClient;
 
     public InfluxDBClient dbConnection() {
-        return InfluxDBClientFactory.create(influxDBProperties.getUrl(), influxDBProperties.getToken().toCharArray(), influxDBProperties.getOrg(), influxDBProperties.getBucket());
+        log.info("url: " + influxDBProperties.getUrl() + " token: " + influxDBProperties.getToken() + " org: " + influxDBProperties.getOrg() + " bucket: " + influxDBProperties.getBucket());
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java]: do we really need this log output?

File: src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java [hunk 10]
```diff
@@ -14,21 +16,35 @@
...
-            try {
-                String receivedMessage = new String(mqttMessage.getPayload());
-                log.info("Received message: " + receivedMessage);
-                if(receivedMessage.contains("1_8_0")) {
-                    SensorEntry sensorEntry = objectMapper.readValue(receivedMessage, SensorEntry.class);
-                    influxDbRepository.writePoint(sensorEntry);
-                }
-            } catch (JsonProcessingException e) {
-                log.error("objectMapper JsonProcessingException" + e);
+    @Override public void messageArrived(String s, MqttMessage mqttMessage) {
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java:23]: method name doesn't really fit to the businnes logic.

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 11]
```diff
@@ -1,50 +1,46 @@
...
+    public void persistPointsToInflux() {
+        log.info("persistPointsToInflux() - every 30 secs");
+        influxDBClient = influxDBConfig.updateConnectionIfNeeded();
+        int numberOfPointsWritten = 0;
+
+        if(influxDBClient.ping()) {
+            log.info("InfluxDbRepository.writePoint() with successful ping()");
+            while (!measurementPoints.isEmpty()) {
+                Point point = measurementPoints.poll();
+                WriteApi writeApi = influxDBClient.makeWriteApi();
```
Comment (burberius) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: Do not create a new Api on every write, that's why it currently dies. See Nasko's but report.

Does the Api also have the ping? Then the DBConfig could return the api instead of the client.

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java (hunk 3 above)
Comment (burberius) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: Check if the queue is not empty, otherwise return. That saves time checking/creating DB connection.

File: src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java (hunk 10 above)
Comment (burberius) [reply src/main/java/de/wejago/hichi2influx/service/SensorMessageSubscriber.java:23]: It is specified from the interface!

**Review by burberius — CHANGES_REQUESTED** [review]

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java (hunk 11 above)
Comment (emiliyank) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: I used the @Bean trick to create it. And from what I read by default it is singleton so should be fine.

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+    @InjectMocks
+    InfluxDBConfig influxDBConfig;
+
+    @BeforeEach
+    public void setup() {
+        when(influxDBProperties.getUrl()).thenReturn(INFLUX_URL);
+        when(influxDBProperties.getOrg()).thenReturn(INFLUX_ORG);
+        when(influxDBProperties.getToken()).thenReturn(INFLUX_TOKEN);
+        doReturn(INFLUX_BUCKET).when(influxDBProperties).getBucket();
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: Didn't the when.thenReturn way work here?
I just wonder, because it looks different and interrupts the reading flow.

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    void testDbConnectionFailure() {
+        //GIVEN
+        when(InfluxDBClientFactory.create(influxDBProperties.getUrl(),
+                                          influxDBProperties.getToken().toCharArray(),
+                                          influxDBProperties.getOrg(),
+                                          influxDBProperties.getBucket()))
+            .thenThrow(new InfluxException(INFLUX_EXCEPTION_MSG));
+
+        InfluxDBClient client = null;
+        try {
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: Use assertThatThrownBy: https://www.baeldung.com/assertj-exception-assertion#1-using-assertthatthrownby

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+    void shouldReturnNullWhenUnableToConnectToInfluxDB(CapturedOutput output) {
+        //GIVEN
+        String INVALID_INFLUX_URL = "invalid-url";
+        when(influxDBProperties.getUrl()).thenReturn(INVALID_INFLUX_URL);
+
+        InfluxDBClient influxDBClient = null;
+
+        try {
+            // WHEN
+            influxDBClient = influxDBConfig.getInfluxDbClient();
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: User assertThatThrownBy

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+        } catch (InfluxException e) {
+            // THEN
+            assertThat(influxDBClient).isNull();
+        }
+    }
+
+    @Test
+    void createInfluxClient_shouldReturnSameClient_whenClientIsNotNullAndPinged() {
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: Can be removed, as it is the same as the first test!
But we need a test for the ping, positive and negative case.

File: src/test/java/de/wejago/hichi2influx/config/InfluxDbConfigWriteApiTest.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    void getWriteApi_returnsNotNull_whenInfluxDBClientIsNotNull() {
+        //GIVEN
+        InfluxDBProperties influxDBProperties = new InfluxDBProperties();
+        influxDBProperties.setOrg(INFLUX_ORG);
+        influxDBProperties.setUrl(INFLUX_URL);
+        influxDBProperties.setBucket(INFLUX_BUCKET);
+        influxDBProperties.setToken(INFLUX_TOKEN);
+        influxDBConfig = new InfluxDBConfig(influxDBProperties);
+        influxDBConfig.getInfluxDbClient();
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDbConfigWriteApiTest.java]: No need to call that, it's done by getWriteApi

File: src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java [hunk 17]
```diff
@@ -16,46 +18,75 @@
...
     @Test
-    void writePointShouldGenerateAndWriteMeasurementPoint() {
+    void shouldPersistPointsToInflux() {
         //GIVEN
-        SensorEntry sensorEntry = generateTestSensorEntry();
-        when(influxDBConfig.dbConnection()).thenReturn(influxDBClient);
-        when(influxDBClient.getWriteApiBlocking()).thenReturn(writeApi);
+        Point point = generateExpectedMeasurementPoint();
+        measurementPoints.add(point);
+        Whitebox.setInternalState(influxDbRepository, "measurementPoints", measurementPoints);
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: You don't need this, you can just write a point and it will be added to the queue.

File: src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java [hunk 18]
```diff
@@ -16,46 +18,75 @@
...
-        Point actualPoint = influxDbRepository.writePoint(sensorEntry);
+        influxDbRepository.persistPointsToInflux();
 
         //THEN
-        Point expectedPoint = generateExpectedMeasurementPoint(sensorEntry);
-        verify(influxDBClient, times(1)).getWriteApiBlocking();
-        assertThat(actualPoint).usingRecursiveComparison()
-            .ignoringFields("time")
-            .isEqualTo(expectedPoint);
+        assertThat(measurementPoints).containsExactly(point);
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: You don't need to check the content of the queue, verifying no write is enough in my opinion.

File: src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java [hunk 19]
```diff
@@ -29,15 +39,15 @@ public class SensorMessageSubscriberTest {
...
+        String receivedMessage = "{\"Time\":\"2023-02-12T21:29:36\",\"SML\":{\"1_8_0\":22417.98}}".trim();
+        SensorEntry expectedSensorEntry = generateTestSensorEntry();
         when(objectMapper.readValue(receivedMessage, SensorEntry.class)).thenReturn(expectedSensorEntry);
 
         // WHEN
         subscriber.messageArrived("test", new MqttMessage(receivedMessage.getBytes()));
 
         // THEN
-        verify(influxDbRepository).writePoint(expectedSensorEntry);
+        verify(influxDbRepository, times(1)).writePoint(any());
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java]: You should check the content of the point to know if the mapping is right.

**Review by burberius — CHANGES_REQUESTED** [review]

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java (hunk 12 above)
Comment (emiliyank) [reply src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: There was a weird issue with the getBucket() and in stackoverflow they suggested to use it this way but didn't work anyway. I put it back to when.thenReturn

File: src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java (hunk 17 above)
Comment (emiliyank) [reply src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: We are testing the persistPointsToInflux() method here. Do you suggest to test the writePoint(point) instead? I'm afraid in this case the code coverage will be very low.

Comment (burberius) [reply src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: You call writePoint(point) to store it in the queue and then persistPointsToInflux() to process the queue.
But you do not have to inject/overwrite the queue from the test.

File: src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java (hunk 19 above)
Comment (emiliyank) [reply src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java]: generateMeasurementPoint(SensorEntry sensorEntry) is a private method. If we want to check the content would be best to make it public, right?

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java (hunk 13 above)
Comment (emiliyank) [reply src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: Here it does not work, the test throws exception - com.influxdb.exceptions.InfluxException: Unable to connect to InfluxDB

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+                                          influxDBProperties.getToken().toCharArray(),
+                                          influxDBProperties.getOrg(),
+                                          influxDBProperties.getBucket()))
+            .thenThrow(new InfluxException(INFLUX_EXCEPTION_MSG));
+
+        InfluxDBClient client = null;
+        //WHEN
+        client = influxDBConfig.getInfluxDbClient();
+
+        assertThatThrownBy(() -> {throw new InfluxException(INFLUX_EXCEPTION_MSG);})
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: That's wrong, you need to test influxDBConfig.getInfluxDbClient()

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+            .isInstanceOf(InfluxException.class)
+            .hasMessage(INFLUX_EXCEPTION_MSG);
+        assertThat(client).isNull();
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java]: not needed

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    void shouldReturnNullWhenUnableToConnectToInfluxDB(CapturedOutput output) {
+        String INVALID_INFLUX_URL = "invalid-url";
+        when(influxDBProperties.getUrl()).thenReturn(INVALID_INFLUX_URL);
+
+        assertThatThrownBy(() -> {
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java:70]: You can remove the {}

File: src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java [hunk 23]
```diff
@@ -1,61 +1,90 @@
...
+        assertThat(measurementPoints).isEmpty();
+        verify(writeApi, never()).writePoint(any());
+    }
+
+    @Test
+    void shouldNotPersistPointsToInfluxWhenWriteApiIsNotAvailable() {
+        //GIVEN
+        Point point = generateExpectedMeasurementPoint();
+        measurementPoints.add(point);
+        Whitebox.setInternalState(influxDbRepository, "measurementPoints", measurementPoints);
```
Comment (burberius) [inline src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: No need for that hack, you can just use the queue that already is part of the influxDbRepository.

**Review by burberius — CHANGES_REQUESTED** [review]

File: src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java (hunk 19 above)
Comment (burberius) [reply src/test/java/de/wejago/hichi2influx/service/SensorMessageSubscriberTest.java]: No, you just create a Point with the content you expect and then use that in the verify instead of any().
Or you use an ArgumentCaptor and assert the result is equal.

File: src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java (hunk 17 above)
Comment (emiliyank) [reply src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java]: I'm not sure if I got what you mean.
But when I remove these 2 lines
```
//        measurementPoints.add(point);
//        Whitebox.setInternalState(influxDbRepository, "measurementPoints", measurementPoints);
```
the test does not work and reason is that the `InfluxDbRepository.measurementPoints` is empty.

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java [hunk 24]
```diff
@@ -1,50 +1,43 @@
...
+        if (!measurementPoints.isEmpty()) {
+            WriteApi writeApi = influxDBConfig.getWriteApi();
+            if (writeApi != null) {
+                int numberOfPointsWritten = 0;
+                while (!measurementPoints.isEmpty()) {
+                    Point point = measurementPoints.poll();
+                    writeApi.writePoint(point);
+                    numberOfPointsWritten++;
+                }
+                log.info("writePoint() wrote : " + numberOfPointsWritten + " point(s).");
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: do we need this log output?

File: src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java [hunk 25]
```diff
@@ -2,15 +2,47 @@
...
+                influxDBClient = getInfluxDbClient();
+            }
+            writeApi = influxDBClient.makeWriteApi();
+        } catch (Exception e) {
+            log.warn("Failed to create writeApi: {}", e.getMessage(), e);
+        }
+    }
+
+    InfluxDBClient getInfluxDbClient() throws InfluxException {
+        log.info("url: " + influxDBProperties.getUrl() + " token: " + influxDBProperties.getToken() + " org: " +
```
Comment (cupakob) [inline src/main/java/de/wejago/hichi2influx/config/InfluxDBConfig.java]: is this log output needed/helpful?

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java (hunk 24 above)
Comment (burberius) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: Perhaps, when it is more than one entry, because then a previous write failed.

File: src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+        InfluxDBClient client = influxDBConfig.getInfluxDbClient();
+
+        //THEN
+        assertThat(client).isNotNull();
+    }
+
+    @Test
+    void testDbConnectionFailure() {
+        when(InfluxDBClientFactory.create(influxDBProperties.getUrl(),
+                                          influxDBProperties.getToken().toCharArray(),
```
Comment (cupakob) [inline src/test/java/de/wejago/hichi2influx/config/InfluxDBConfigTest.java:55]: try to keep `// GIVEN`, `// WHEN`, `// THEN` if possible

File: src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java (hunk 24 above)
Comment (cupakob) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: disagree - then you will see, that more than one point was wrote, but this is not really valuable information. And in the regular case you will see only `wrote 1 point(s)` and this is not helpful. And on the other side this will generate a lot of log outputs. Maybe this is helpful only for debug porposes.

Comment (burberius) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: You see that there was something wrong before and it now reconnected and wrote X points.
Later, when it worked for some time, we can remove it.

Comment (emiliyank) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: We did something in the middle - we will log X points only when X > 1.

Comment (burberius) [reply src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java]: That's what I wrote ;-P

File: deployment/desktop.ini
Comment (burberius) [inline deployment/desktop.ini:1]: That file is probably from Windows and shouldn't be checked in

File: et --soft HEAD~3 &&
Comment (burberius) [inline et --soft HEAD~3 &&:1]: What's that???

File: deployment/mqtt2influx-configuration.yaml
Comment (burberius) [inline deployment/mqtt2influx-configuration.yaml:1]: Should be in another PR for the other ticket

**Review by burberius — CHANGES_REQUESTED** [review]

File: et --soft HEAD~3 &&
Comment (emiliyank) [reply et --soft HEAD~3 &&:1]: seems it was generated while I was trying to do squash

File: deployment/mqtt2influx-configuration.yaml
Comment (emiliyank) [reply deployment/mqtt2influx-configuration.yaml:1]: removed from this branch

**Review by cupakob — APPROVED** [review]

**Review by burberius — APPROVED** [review]
